tests/async/async_ref_test.cc

```diff
@@ -122,236 +151,126 @@ TEST(AsyncRef, AddUpdateAndRemove) {
 
-			ASSERT_EQ(comp->value_to_update, 6);
-		};
+	evc.set_update_callback<async_test::ComponentUpdate>(
+		[&](ecsact_entity_id id, const async_test::ComponentUpdate& component) {
+			update_happened = true;
+			ASSERT_EQ(component.value_to_update, 6);
+		}
+	);
 
-	// Prepare update component data
 	my_update_component.value_to_update += 5;
 
-	// Update components
-	options.update_component(cb_info.entity, &my_update_component);
+	options.update_component(entity, &my_update_component);
 
 	ecsact_async_enqueue_execution_options(options.c());
 
-	start_tick = ecsact_async_get_current_tick();
-	while(!cb_info.update_happened) {
-		std::this_thread::yield();
-		ecsact_async_flush_events(&evc, nullptr);
-		auto current_tick = ecsact_async_get_current_tick();
-		auto tick_diff = current_tick - start_tick;
-		ASSERT_LT(tick_diff, 10);
-	}
+	c_events_collector = evc.c();
+
+	flush_with_condition(update_happened, c_events_collector);
 
 	options.clear();
-	evc.update_callback = {};
-	evc.update_callback_user_data = nullptr;
+	evc.clear();
+
+	bool remove_happened = false;
 
 	// Remove component
-	options.remove_component<async_test::ComponentUpdate>(cb_info.entity);
+	options.remove_component<async_test::ComponentUpdate>(entity);
 
 	ecsact_async_enqueue_execution_options(options.c());
 
-	evc.remove_callback_user_data = &cb_info;
-	evc.remove_callback = //
-		[](
-			ecsact_event        event,
-			ecsact_entity_id    entity_id,
-			ecsact_component_id component_id,
-			const void*         component_data,
-			void*               callback_user_data
-		) {
-			auto& cb_info = *static_cast<callback_data*>(callback_user_data);
-
-			cb_info.remove_happened = true;
-			ASSERT_EQ(component_id, async_test::ComponentUpdate::id);
-
-			auto comp =
-				static_cast<const async_test::ComponentUpdate*>(component_data);
+	evc.set_remove_callback<async_test::ComponentUpdate>(
+		[&](ecsact_entity_id id, const async_test::ComponentUpdate& component) {
+			remove_happened = true;
+			ASSERT_EQ(component.value_to_update, 6);
+		}
+	);
 
-			ASSERT_EQ(comp->value_to_update, 6);
-		};
+	c_events_collector = evc.c();
 
-	start_tick = ecsact_async_get_current_tick();
+	auto start_tick = ecsact_async_get_current_tick();
 	// Flush until we hit a maximum or get all our events we expect.
-	while(!cb_info.all_events_happened()) {
+	while(init_happened && update_happened && remove_happened) {
 		std::this_thread::yield();
-		ecsact_async_flush_events(&evc, nullptr);
+		ecsact_async_flush_events(&c_events_collector, nullptr);
 		auto current_tick = ecsact_async_get_current_tick();
 		auto tick_diff = current_tick - start_tick;
-		ASSERT_LT(tick_diff, 10)
-			<< "Not all events happened before maximum was reached\n"
-			<< "  init = " << cb_info.init_happened << "\n"
-			<< "  update = " << cb_info.update_happened << "\n"
-			<< "  remove = " << cb_info.remove_happened << "\n";
+		ASSERT_LT(tick_diff, 10) << "Not all events happened before maximum was "
```

Comment:
If you want this to format on each line you can add a `//` before the `<<` and put `<<` on the next line
```suggestion
		ASSERT_LT(tick_diff, 10) //
		 << "Not all events happened before maximum was reached\n"
```